### DISCUSSION TITLE
Cannot set property 'styles' of undefined

### DISCUSSION THREAD

**Discussion by mtahagocer** [vercel/next.js#27026] [Help] [answered]
### What version of Next.js are you using?

11.0.1

### What version of Node.js are you using?

16.4.1

### What browser are you using?

chrome

### What operating system are you using?

windows

### How are you deploying your application?

vercel

### Describe the Bug

I'm trying to use antd on nextjs project but with less options its really getting crazy.

my next.config.js

`const withSass = require('@zeit/next-sass');
const withLess = require('@zeit/next-less');
const withCSS = require('@zeit/next-css');

if (typeof require !== 'undefined') {
  require.extensions['.less'] = (file) => { };
}

module.exports = withCSS({
  cssModules: true,
  cssLoaderOptions: {
    importLoaders: 1,
    localIdentName: '[local]___[hash:base64:5]',
  },
  ...withLess(
    withSass({
      lessLoaderOptions: {
        javascriptEnabled: true,
      },
    })
  ),
});
`

```
Then I run next dev

$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from D:\Projects\powerdev-frontend\.env.local      
info  - Loaded env from D:\Projects\powerdev-frontend\.env.development
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
Warning: Built-in CSS support is being disabled due to custom CSS configuration being detected.
See here for more info: https://nextjs.org/docs/messages/built-in-css-disabled

TypeError: Cannot set property 'styles' of undefined
    at module.exports (D:\Projects\powerdev-frontend\node_modules\@zeit\next-css\css-loader-config.js:25:56)
    at Object.webpack (D:\Projects\powerdev-frontend\node_modules\@zeit\next-css\index.js:15:36)
    at getBaseWebpackConfig (D:\Projects\powerdev-frontend\node_modules\next\dist\build\webpack-config.js:162:454)    
    at async Promise.all (index 0)
    at async HotReloader.start (D:\Projects\powerdev-frontend\node_modules\next\dist\server\hot-reloader.js:16:133)   
    at async DevServer.prepare (D:\Projects\powerdev-frontend\node_modules\next\dist\server\next-dev-server.js:16:453)
    at async D:\Projects\powerdev-frontend\node_modules\next\dist\cli\next-dev.js:22:1
error Command failed with exit code 1.
```

### Expected Behavior

İt must be de easy to use any css type what with nextjs its really going bad.

### To Reproduce

Try to use antd with next lates and see what happen

**Comment by agreco** [comment]
Had the same error when trying to upgrade with Next 11 and its default use with Webpack 5. 

I found this: https://nextjs.org/docs/messages/webpack5 and added the `webpack5: false` flag to my next.config.js, the error doesn't occur anymore. Obviously you lose the benefits that come with Webpack 5, but for now this is a viable workaround.

I use the @zeit/next-less, @zeit/next-css and antd packages, so obviously there's a compatibility issue with some or all of these packages. I've tried upgrading these to their latest versions, which at the time of writing are 1.0.1, 1.0.1 and 4.16.11 respectively, but to no avail. 

Note: @zeit/next-less has now been archived, so I don't think there will be more releases now if it is indeed causing this issue. I ran a quick search and found https://github.com/SolidZORO/next-plugin-antd-less, which I'm going to try and migrate to. Also if this PR https://github.com/vercel/next.js/pull/23185 gets merged then Next will support Less out of the box.

**Comment by mtahagocer** [comment, marked as answer]
Sory for late,  solution is [here](https://www.npmjs.com/package/less-vars-to-js)

**Comment by agreco** [comment]
@mtahagocer Thanks for the heads up on the solution. Would you mind giving some instructions on how to apply less-vars-to-js in a setup within next.config.js, in particular as a replacement for @zeit/next-less please?

**Reply by mtahagocer** [reply]
There is a few packages like [this](https://www.npmjs.com/package/next-plugin-antd-less), fixes "less" problem, but they are really have trouble on packages which are using "less" like antd, so there is no **real** solution for "less" on nextjs, sory.

**Reply by agreco** [reply]
Ah ok, thanks for elaborating. Regarding the solution you linked to, `less-vars-to-js`, how can it be used to support Less in Next?

**Reply by mtahagocer** [reply]
Check here https://www.npmjs.com/package/next-with-less,
using less-vars-to-js for on the "modifyVars" parameters which on documentation, you can read all your theme variables with file stream and give it to less-vars-to-js for json variables, then u can declare "modifyVars" with programmatic solution
